tests/unit/test_jenkins.py

```diff
@@ -1039,59 +1039,6 @@ def test__build_dependencies_lookup(
     assert lookup == expected_lookup
 
 
-@pytest.mark.parametrize(
-    "plugin, plugins_lookup, expected_plugins",
-    [
-        pytest.param(
-            "plugin-a",
-            {
-                "plugin-a": ("plugin-b", "plugin-c"),
-                "plugin-b": ("plugin-d",),
-                "plugin-c": (),
-                "plugin-d": (),
-            },
-            ("plugin-a", "plugin-b", "plugin-d", "plugin-c"),
-            id="no duplicate dependencies",
-        ),
-        pytest.param(
-            "plugin-a",
-            {
-                "plugin-a": ("plugin-b", "plugin-c"),
-                "plugin-b": ("plugin-d", "plugin-c"),
-                "plugin-c": ("plugin-d"),
-                "plugin-d": (),
-            },
-            ("plugin-a", "plugin-b", "plugin-d", "plugin-c"),
-            id="duplicate dependencies",
-        ),
-        pytest.param(
-            "plugin-a",
-            {
-                "plugin-a": ("plugin-b", "plugin-e"),
-                "plugin-b": ("plugin-d", "plugin-e"),
-                "plugin-c": ("plugin-d"),
-                "plugin-d": (),
-            },
-            ("plugin-a", "plugin-b", "plugin-d"),
-            id="dependency not in lookup table",
-        ),
-    ],
-)
-def test__traverse_dependencies(
-    plugin: str,
-    plugins_lookup: typing.Mapping[str, typing.Iterable[str]],
-    expected_plugins: typing.Tuple[str],
-):
-    """
-    arrange: given the plugins lookup table and a target plugin to recursively find dependencies.
-    act: when _traverse_dependencies is called.
-    assert: the plugin and its dependencies are traversed.
-    """
-    traversed_plugins = jenkins._traverse_dependencies(plugin, plugins_lookup, set())
-
-    assert tuple(traversed_plugins) == expected_plugins
-
-
 @pytest.mark.parametrize(
     "top_level_plugins, plugins_lookup, expected_allowed_plugins",
     [
```

Comment:
For test cases, I would suggest ordering them logically. For example:

- all empty
- one plugin in top level and
  - nothing in lookup
  - one in lookup but its different
  - one in lookup and it matches the plugin with
    - no extra dependencies
    - one dependency and
      - no nested depedency
      - single nested dependency
    - multiple dependencies
- two plugins where
  - none of them have dependencies
  - one of them has and the other doesn't have dependencies (2 test cases here)
  - both of them have a single dependency and
    - the dependencies are different
    - they are the same
  - both of them have multiple dependencies and
    - there is no overlap
    - one is the same (2 test cases here)
    - both are the same

Let's not use a set so that we can check whether the seen plugins check works